tests/flytekit/unit/tools/test_script_mode.py

```diff
@@ -237,3 +242,71 @@ def test_get_all_modules(tmp_path):
     # Workflow exists, so it is imported
     workflow_file.write_text(WORKFLOW_CONTENT)
     assert n_sys_modules + 1 == len(get_all_modules(os.fspath(source_dir), "my_workflows.main"))
+
+
+def test_list_imported_modules_as_files():
+
+    bin_directory = Path(os.path.dirname(sys.executable))
+    flytekit_root = Path(os.path.dirname(flytekit.__file__))
+    root_dir = Path("C:\\") if platform.system() == "Windows" else Path("/")
+    source_path = root_dir / "home" / "username" / "project"
+
+    site_packages = [
+        str(source_path / ".venv" / "lib" / "python3.10" / "site-packages"),
+        str(source_path / ".venv" / "local" / "lib" / "python3.10" / "dist-packages"),
+        str(source_path / ".venv" / "lib" / "python3" / "dist-packages"),
+        str(source_path / ".venv" / "lib" / "python3.10" / "dist-packages"),
+    ]
+
+    # Setup mock module objects
+    # Local modules that should be included
+    local_modules = [
+        ModuleType("local_module_1"),
+        ModuleType("local_module_2"),
+        ModuleType("local_module_3"),
+        ModuleType("local_module_4")
+    ]
+    local_module_paths = [
+        str(source_path / "package_a" / "module_1.py"),
+        str(source_path / "package_a" / "module_2.py"),
+        str(source_path / "package_b" / "module_3.py"),
+        str(source_path / "package_b" / "module_4.py"),
+    ]
+    # Flyte module that should be excluded
+    flyte_modules = [
+        ModuleType("flyte_module")
+    ]
+    flyte_module_paths = [
+        str(flytekit_root / "package" / "module.py"),
+    ]
+    # bin module that should be excluded
+    bin_modules = [
+        ModuleType("bin_module")
+    ]
+    bin_module_paths = [
+        str(bin_directory / "package" / "module.py"),
+    ]
+    # site modules that should be excluded
+    site_modules = [
+        ModuleType("site_module_1"),
+        ModuleType("site_module_2"),
+        ModuleType("site_module_3"),
+        ModuleType("site_module_4"),
+    ]
+    site_module_paths = [
+        str(Path(site_packages[0]) / "package" / "module_1.py"),
+        str(Path(site_packages[1]) / "package" / "module_2.py"),
+        str(Path(site_packages[2]) / "package" / "module_3.py"),
+        str(Path(site_packages[3]) / "package" / "module_4.py"),
+    ]
+
+    modules = local_modules + flyte_modules + bin_modules + site_modules
+    module_paths = local_module_paths + flyte_module_paths + bin_module_paths + site_module_paths
+
+    for m, p in zip(modules, module_paths):
+        m.__file__ = p
```

Comment:
Depending on two list being aligned feels a little brittle. Can have the modules be a list of tuples? Something like this:

```python
local_modules = [
    (ModuleType("local_module_1"), str(source_path / "package_a" / "module_1.py")),
    ...
]

all_modules = local_modules + flyte_modules + ...

for mod, mod_path in all_modules:
    mod.__file__  = mod_path

# to be passed in later
modules = [mod for mod, _ in all_modules]

with mock.patch("site.getsitepackages", new=lambda: site_packages):
    file_list = list_imported_modules_as_files(str(source_path), modules)

local_module_paths = [path for _, path in local_modules]
assert sorted(file_list) == sorted(local_module_paths)
```